apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/modal.js

```diff
@@ -3,26 +3,72 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { Button, Modal } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { useEffect } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import React from 'react';
 import image from './image.svg';
 
 import './modal.scss';
 
 const GlobalStylesModal = () => {
+	const params = new URLSearchParams( window.location.search );
+	const [ showUnlockStylesModal, setShowUnlockStylesModal ] = useState(
+		params.get( 'unlock-styles' ) && wpcomGlobalStyles.inUse
+	);
+
 	const isVisible = useSelect(
-		( select ) => select( 'automattic/wpcom-global-styles' ).isModalVisible(),
-		[]
+		( select ) => {
+			if ( ! showUnlockStylesModal ) {
+				return select( 'automattic/wpcom-global-styles' ).isModalVisible();
+			}
+
+			// Show unlock styles modal on load, but wait for the global styles to be ready.
+			const { getEditedPostId, getEditedPostType } = select( 'core/edit-site' );
+			if ( getEditedPostType() === undefined || getEditedPostId() === undefined ) {
+				return false;
+			}
```

Comment:
This is just a question. Why do we need to wait for global styles to be ready?

Reply:
Because otherwise the modal will show up before the editor is visible.

<img width="1278" alt="Screen Shot 2022-10-26 at 17 59 41" src="https://user-images.githubusercontent.com/1233880/198076313-f79b104a-5108-4897-8761-5d25f8baae76.png">

Reply:
Is there a simpler condition for checking if the editor is visible? I have the feeling we are checking a lot of conditions. Does Gutenberg have a store with the editor's status, I find it confusing that we tie global styles being loaded to the editor being loaded.

Reply:
> Is there a simpler condition for checking if the editor is visible?

No 😢. I wish Gutenberg could expose these `isReady` properties (which is the one that prevents the editor from being rendered) but there are not:

https://github.com/WordPress/gutenberg/blob/aa55590ea7eb16cb90dd41531ff10c86d52023c2/packages/edit-site/src/components/global-styles/global-styles-provider.js#L140

https://github.com/WordPress/gutenberg/blob/aa55590ea7eb16cb90dd41531ff10c86d52023c2/packages/edit-site/src/components/editor/index.js#L183-L186